fail2ban/server/filter.py

```diff
@@ -844,35 +943,50 @@ def getJournalMatch(self, match): # pragma: no cover - Base class, not used
 
 import socket
 import struct
+from .utils import Utils
 
 
 class DNSUtils:
 
 	IP_CRE = re.compile("^(?:\d{1,3}\.){3}\d{1,3}$")
 
+	# todo: make configurable the expired time and max count of cache entries:
+	CACHE_dnsToIp = Utils.Cache(maxCount=1000, maxTime=60*60)
```

Comment:
hm, I also was somewhat skeptical about DNS caching advantage, and thought that core python/OS provides some noteable one BUT apparently not quite so :
```
*In [16]: %timeit -n1 -r1 socket.gethostbyname_ex("thebest.com")
1 loops, best of 1: 38.2 ms per loop

*In [17]: %timeit socket.gethostbyname_ex("thebest.com")
10 loops, best of 3: 20.6 ms per loop
*In [18]: %timeit -n1 -r1 socket.gethostbyname_ex("aloha.com")
1 loops, best of 1: 158 ms per loop

*In [19]: %timeit socket.gethostbyname_ex("aloha.com")
100 loops, best of 3: 6.93 ms per loop
```
so caching is in place BUT it is quite "heavy" and fail2ban-side caching could indeed be of effect, although I am yet not sure why there should be much of DNS querying for typical services filters of which should rely on catching IPs, not DNS names/...  ???